src/apps/monero/protocol/signing/step_01_init_transaction.py

```diff
@@ -0,0 +1,242 @@
+"""
+Initializes a new transaction.
+"""
+
+import gc
+
+from apps.monero.controller import misc
+from apps.monero.layout import confirms
+from apps.monero.protocol.signing.state import State
+from apps.monero.xmr import common, crypto, monero
+
+
+async def init_transaction(state: State, address_n, network_type, tsx_data):
+    from apps.monero.xmr.sub.addr import classify_subaddresses
+    from apps.monero.protocol import hmac_encryption_keys
+
+    state.creds = await misc.monero_get_creds(state.ctx, address_n, network_type)
+
+    state.fee = state.fee if state.fee > 0 else 0
+
+    state.tx_priv = crypto.random_scalar()
+    state.tx_pub = crypto.scalarmult_base(state.tx_priv)
+
+    state.mem_trace(1)
+
+    # Ask for confirmation
+    await confirms.confirm_transaction(state.ctx, tsx_data, state.creds)
+    gc.collect()
+    state.mem_trace(3)
+
+    # Basic transaction parameters
+    state.input_count = tsx_data.num_inputs
+    state.output_count = len(tsx_data.outputs)
+    state.output_change = tsx_data.change_dts
+    state.mixin = tsx_data.mixin
+    state.fee = tsx_data.fee
+    state.account_idx = tsx_data.account
+    state.multi_sig = tsx_data.is_multisig
+    check_change(state, tsx_data.outputs)
+    state.exp_tx_prefix_hash = tsx_data.exp_tx_prefix_hash
+
+    # Rsig data
+    state.rsig_type = tsx_data.rsig_data.rsig_type
+    state.rsig_grp = tsx_data.rsig_data.grouping
+    state.rsig_offload = state.rsig_type > 0 and state.output_count > 2
+    state.use_bulletproof = state.rsig_type > 0
+    state.use_simple_rct = state.input_count > 1 or state.rsig_type != 0
```

Comment:
I'd like to make it here a little bit clearer. There are four options how to make the range signature:
1. borromean offloaded
1. borromean in trezor
1. bulletproof offloaded
1. bulletproof in trezor

Am I correct? And "simple" means in trezor, "full" offloaded?

If the above is correct `rsig_type` is equal to 0 if borromean and 1 if bulletproofs?

Reply:
Simple RCT is a standalone mode how MGsig ring is constructed from mixin. Used iff num_inputs == 1.

Borromean offloading is not currently implemented as we don't need it now. All other 3 options are used. 

rsig_type 0 = borromean, 1 = bulletproof

Reply:
To be more precise, >=1 is Bulletproof.

```
enum RangeProofType { RangeProofBorromean, RangeProofBulletproof, RangeProofMultiOutputBulletproof, RangeProofPaddedBulletproof };
```

https://github.com/monero-project/monero/blob/83d8f03c2338e865ce72f4e2d46d43a48530ab10/src/ringct/rctTypes.h#L234

Reply:
Ok, let me summarize. There are two basic types of Ring Confidential Transactions (RCT):

1. RCTTypeFull
2. RCTTypeSimple

`state.use_simple_rct` denotes which one of those two is used.

For those two transaction types we have two (well, three) range proof types:
1. borromean
2. bulletproof in trezor
3. bulletproof offloaded

Is that correct?

Reply:
Btw the [Zero to Monero](https://getmonero.org/library/Zero-to-Monero-1-0-0.pdf) book says it is the opposite, doesn't it?

> For multi-input transactions, existing Monero wallets use the RCTTypeSimple..

> Our perception is that the decision to limit RCTTypeFull transactions to one input..

Reply:
Thats correct, source code has it right. Simple is for multi-inputs.

Reply:
And I think the Full version is only used with Borromean rsig

Reply:
Ok, I think I'll just put it in a separate function for now and we can tackle it later to make it a little bit clearer.